packages/runtime/container-runtime/src/gc/gcUnreferencedStateTracker.ts

```diff
@@ -45,54 +47,90 @@ export class UnreferencedStateTracker {
 		private readonly inactiveTimeoutMs: number,
 		/** The current reference timestamp used to track how long this node has been unreferenced for. */
 		currentReferenceTimestampMs: number,
-		/** The time after which node transitions to SweepReady state; undefined if session expiry is disabled. */
-		private readonly sweepTimeoutMs: number | undefined,
+		/** The time after which node transitions to TombstoneReady state; undefined if session expiry is disabled. */
+		private readonly tombstoneTimeoutMs: number | undefined, //* Rename to tombstoneTimeoutMs
+		/** The delay from TombstoneReady to SweepReady (only applies if tombstoneTimeoutMs is defined) */
+		private readonly tombstoneSweepDelayMs: number,
 	) {
-		if (this.sweepTimeoutMs !== undefined) {
-			assert(
-				this.inactiveTimeoutMs <= this.sweepTimeoutMs,
-				0x3b0 /* inactive timeout must not be greater than the sweep timeout */,
-			);
-		}
+		assert(
+			this.tombstoneTimeoutMs === undefined ||
+				this.tombstoneTimeoutMs >= this.inactiveTimeoutMs,
+			0x3b0 /* inactiveTimeoutMs must not be greater than the tombstoneTimeoutMs */,
+		);
 
 		this.sweepTimer = new TimerWithNoDefaultTimeout(() => {
 			this._state = UnreferencedState.SweepReady;
+			assert(
+				!this.inactiveTimer.hasTimer && !this.tombstoneTimer.hasTimer,
+				0x3b1 /* inactiveTimer or tombstoneTimer still running after sweepTimer fired! */,
+			);
+		});
+
+		this.tombstoneTimer = new TimerWithNoDefaultTimeout(() => {
+			this._state = UnreferencedState.TombstoneReady;
 			assert(
 				!this.inactiveTimer.hasTimer,
-				0x3b1 /* inactiveTimer still running after sweepTimer fired! */,
+				0x3b1 /* inactiveTimer still running after tombstoneTimer fired! */,
```

Comment:
I don't think you can update the string for an existing assert code.

Reply:
I was wondering about that.  Oh, and anyway I should not duplicate the assert code above.  Maybe I'll clear them both.

Reply:
@alexvy86 - If I want to update the message for an existing assert, is it better to just delete the code and have a new one generated?  Or can the system handle the update?

Reply:
I just removed the codes, will generate new ones